lib/openshift/machine.rb

```diff
@@ -43,6 +43,12 @@ def ready?(user: nil, cached: true, quiet: false)
       instance_state == 'running'
     end
 
+    #location  for Azure machines set by installer 
+    def location(user: nil, cached: true, quiet: false)
```

Comment:
This property seems only valid for Azure. Perhaps rename it to `azure_location`. In case all or most  clouds have such property, then I agree to keep it as is and it will return the location depending on which cloud it is in.